projects/batfish-client/src/main/java/org/batfish/client/Client.java

```diff
@@ -2606,6 +2610,66 @@ private boolean showVersion(List<String> options, List<String> parameters) {
     return true;
   }
 
+  private boolean syncTestrigsSyncNow(List<String> options, List<String> parameters) {
+    if (!isValidArgument(options, parameters, 1, 1, 1,
+            Command.SYNC_TESTRIGS_SYNC_NOW)) {
+      return false;
+    }
+    if (!isSetContainer(true)) {
+      return false;
+    }
+
+    boolean force = false;
+
+    if (options.size() == 1) {
+      if (options.get(0).equals("-force")) {
+        force = true;
+      } else {
+        _logger.errorf("Unknown option: %s\n", options.get(0));
+        printUsage(Command.SYNC_TESTRIGS_SYNC_NOW);
+        return false;
+      }
+    }
+
+    String pluginId = parameters.get(0);
+
+    return _workHelper.syncTestrigsSyncNow(pluginId, _currContainerName, force);
+  }
+
+  private boolean syncTestrigsUpdateSettings(String[] words, List<String> options,
+                                             List<String> parameters) {
+    if (!isValidArgument(options, parameters, 0, 1, Integer.MAX_VALUE,
+            Command.SYNC_TESTRIGS_UPDATE_SETTINGS)) {
+      return false;
+    }
+    if (!isSetContainer(true)) {
+      return false;
+    }
+
+    String pluginId = parameters.get(0);
+
+    String settingsStr = "{"
+            + String.join(" ", Arrays.copyOfRange(words, 2 + options.size(), words.length))
+            + "}";
+
+    Map<String, String> settings = null;
+
+    try {
+      BatfishObjectMapper mapper = new BatfishObjectMapper();
+      settings = mapper.readValue(new JSONObject(settingsStr).toString(),
```

Comment:
what does `new JSONObject(string).toString()` do to the string?

Reply:
The parser that JSONObject uses is more liberal (and easier for humans to handle). We use that to parse the user supplied input and then the toString() function spits out a "proper" JSON string that mapper can parse. (We use this pattern in another place.)